plugins/org.locationtech.udig.project.ui/src/org/locationtech/udig/project/ui/internal/commands/draw/DrawCoordinateCommand.java

```diff
@@ -0,0 +1,244 @@
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.operation.TransformException;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+
+/**
+ * Draws a Coordinate at the specified location on the screen viewport.
+ * 
+ * @author nprigour
+ * 
+ * @since 2.0
+ */
+public class DrawCoordinateCommand extends AbstractDrawCommand {
+
+    private CoordinateReferenceSystem crs;
+
+    private Coordinate coord;
+
+
+    private Color paint = null;
+
+    private int style = -1;
+
+    private int width;
+
+    private Color fill;
+
+    private boolean useCircle;
+    
+    private boolean errorReported;
+    
+    
+    /**
+     * @param coord
+     * @param map Map to consider as active for drawing
+     */
+    public DrawCoordinateCommand(Coordinate coord, IMap map) {
+        this(coord, map.getEditManager().getSelectedLayer().getCRS());
+        setMap(map);
+    }
+
+    /**
+     * @param coord
+     * @param crs
+     */
+    public DrawCoordinateCommand(Coordinate coord, CoordinateReferenceSystem crs) {
+        this.coord = coord;
+        if (crs == null) {
+            this.crs = DefaultGeographicCRS.WGS84;
+        } else {
+            this.crs = crs;
+        }
+        useCircle = false;
+    }
+
+    /**
+     * Does the actual job by drawing a 3x3 rectangle or circle at the 
+     * coordinate location
+     * 
+     * @see org.locationtech.udig.project.command.MapCommand#run()
+     */
+    public void run(IProgressMonitor monitor) throws Exception {
+
+        final Point point = getMap().getViewportModel().worldToPixel(coord);
+        Shape shape = useCircle ? new Ellipse2D.Double(point.x, point.y, 3, 3) : 
+            new Rectangle2D.Double(point.x, point.y, 3, 3);
+        
+        if (fill != null) {
+            graphics.setColor(fill);
+            graphics.fill(shape);
+        }
+
+        if (paint != null)
+            graphics.setColor(paint);
+        if (style > -1)
+            graphics.setStroke(style, width);
+        doDraw(shape);
+    }
+
+    /**
+     * actually the Shape is always of type Rectangle2D but we may enhance 
+     * and make it configurable in the future.
+     *  
+     * @param shape
+     */
+    private void doDraw(Shape shape) {
+        if (shape instanceof Rectangle2D) {
+            Rectangle2D rect = (Rectangle2D) shape;
+            graphics.drawRect((int) rect.getX(), (int) rect.getY(), (int) rect.getWidth(),
+                    (int) rect.getHeight());
+            return;
+        }
+
+        if (shape instanceof Ellipse2D) {
+            Ellipse2D ellipse = (Ellipse2D) shape;
+            graphics.drawOval((int) ellipse.getMinX(), (int) ellipse.getMinY(),
+                    (int) ellipse.getWidth(), (int) ellipse.getHeight());
+            return;
+        }
+    }
+
+    /**
+     * @return Returns the paint.
+     */
+    public Paint getPaint() {
```

Comment:
Is this the paintColor for the Coordinate? I guess because internally its a Color object it should return Color here as well. I generall I prefer members to name what it is for, ig. paintColor with named getters and setters (like you almost did for the lineStyle ;))

Reply:
OK I will fix it.